plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -496,6 +497,38 @@ private void predicatePushdownTest(String oracleType, String oracleLiteral, Stri
         }
     }
 
+    @Test
+    public void testJoinPushdownWithImplicitCast()
+    {
+        try (TestTable leftTable = new TestTable(getQueryRunner()::execute, "left_table", "(id int, varchar_50 varchar(50))", ImmutableList.of("(1, 'India')", "(2, 'Poland')"));
+                TestTable rightTable = new TestTable(getQueryRunner()::execute, "right_table_", "(varchar_100 varchar(100), varchar_unbounded varchar)", ImmutableList.of("('India', 'Japan')", "('France', 'Poland')"))) {
+            String leftTableName = leftTable.getName();
+            String rightTableName = rightTable.getName();
+            Session session = joinPushdownEnabled(getSession());
+
+            // Implict cast between bounded varchar
+            assertJoin(session, leftTableName, rightTableName, "LEFT JOIN", "l.varchar_50 = r.varchar_100")
```

Comment:
@ssheikin Would have a different opinion - But I would like to see the query being inlined here

Reply:
In this case inlining of method is justified. ..but `"SELECT id FROM %s l %s %s r ON %s"` may be extracted as variable 😄